api/document/js/Api.ts

```diff
@@ -18,31 +18,60 @@ export function setStatusError(e: Error) {
   setStatus(errMsg, 'editor-status-error');
 }
 
-export default class Api {
+export interface ApiNode {
+  children: ApiNode[];
+  content: ApiContent[];
+  marker?: string;
+  node_type: string;
+  type_emblem?: string;
+  title?: string;
+  identifier?: string;
+  text?: string;
+}
+
+export interface ApiContent {
+  content_type: string;
+  inlines: ApiContent[];
+  text: string;
+}
+
+interface ApiTypeMap {
+  'json': ApiNode;
+  'akn+xml': string;
+}
+
+interface ApiOptions<T extends keyof ApiTypeMap> {
+  contentType: T;
+  csrfToken: string;
+  url: string;
+}
```

Comment:
IIRC, typed object destructuring isn't supported very well in TypeScript, so I've just split `ApiOptions` out into its own separate interface for now.  Not ideal but I'm not sure how else to do this.